### PR TITLE
Add TimeDeltaDatetime format class for datetime.timedelta

### LINKED ISSUE #4982: astropy.time.TimeDelta should support conversion to `datetime.timedelta`
Because of the inheritance `TimeDelta` has a method `to_datetime` which is useless.

It should have a method `to_timedelta` which returns a `datetime.timedelta` object or objects.

Conversion to `np.datetime64` (for `Time`) and `np.timedelta64` (for `TimeDelta`) would also be great.

### PR DESCRIPTION
**Pull request by vn-ki**
`TimeDelta` now can be initialized and can be converted to `datetime.timedelta`. Overridden `to_datetime` in `TimeDelta` to return `timedelta` and ignore `timezone`.

Closes #4982.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (bsipocz) [inline .vscode/settings.json]: please remove this file, you can rather add `.vscode` to the main `.gitignore` file.

**Comment by vn-ki** [comment]
@taldcroft Good to know that you didn't find any real issues. :champagne:

**Comment by bsipocz** [comment]
Travis failure is unrelated, and this has passed before already so no need to restart.

@vn-ki  - Please use the `[skip ci]` tag in the commit message for minor otherwise untested commits (like adding a changelog entry) as there is no point in running CI for those.

**Comment by taldcroft** [comment]
Sorry, I missed one thing which is that it needs an update to the narrative docs to add the new format:
http://astropy.readthedocs.io/en/latest/time/index.html#time-deltas

File: astropy/time/core.py [hunk 2]
```diff
@@ -1773,6 +1773,9 @@ def replicate(self, *args, **kwargs):
         return out
 
     def to_datetime(self):
+        """
+        Convert to `datetime.timedelta` object.
```
Comment (bsipocz) [inline astropy/time/core.py]: I think this will fail to be converted to a link, so use double backticks instead.

Comment (vn-ki) [reply astropy/time/core.py]: `~datetime.datetime` worked in the `to_datetime` of `Time`. That's why I added.

Comment (bsipocz) [reply astropy/time/core.py]: oh, ok, then sorry for the noise here.

Comment (vn-ki) [reply astropy/time/core.py]: But I updated. Should I revert it back?

**Comment by taldcroft** [comment]
Thanks @vn-ki this is a nice addition!

**Comment by vn-ki** [comment]
Thank you for the merge!!

**Comment by Cadair** [comment]
:fireworks:

**Pull request merged**